containerd-shim-spin/src/engine.rs

```diff
@@ -152,47 +155,52 @@ impl SpinEngine {
     ) -> Result<()> {
         let mut futures_list = Vec::new();
         let mut trigger_type_map = Vec::new();
-
+        let mut loader = spin_trigger::loader::ComponentLoader::default();
+        match app_source {
+            Source::Oci => unsafe {
+                // Configure the loader to support loading AOT compiled components..
+                // Since all components were compiled by the shim (during `precompile`),
+                // this operation can be considered safe.
+                loader.enable_loading_aot_compiled_components();
+            },
+            // Currently, it is only possible to precompile applications distributed using
+            // `spin registry push`
+            Source::File(_) => {}
+        };
         for trigger_type in trigger_types.iter() {
+            let app = spin_app::App::new("TODO", app.clone());
             let f = match trigger_type.as_str() {
-                HttpTrigger::TRIGGER_TYPE => {
-                    let http_trigger =
-                        build_trigger::<HttpTrigger>(app.clone(), app_source.clone()).await?;
+                HTTP_TRIGGER_TYPE => {
                     info!(" >>> running spin http trigger");
                     let address_str = env::var(constants::SPIN_HTTP_LISTEN_ADDR_ENV)
                         .unwrap_or_else(|_| constants::SPIN_ADDR_DEFAULT.to_string());
                     let address = parse_addr(&address_str)?;
-                    http_trigger.run(spin_trigger_http::CliArgs {
+                    let cli_args = spin_trigger_http::CliArgs {
                         address,
                         tls_cert: None,
                         tls_key: None,
-                    })
+                    };
+                    trigger::run::<HttpTrigger>(cli_args, app, &loader).await?
                 }
-                RedisTrigger::TRIGGER_TYPE => {
-                    let redis_trigger =
-                        build_trigger::<RedisTrigger>(app.clone(), app_source.clone()).await?;
+                REDIS_TRIGGER_TYPE => {
                     info!(" >>> running spin redis trigger");
-                    redis_trigger.run(spin_trigger::cli::NoArgs)
+                    trigger::run::<RedisTrigger>(NoCliArgs, app, &loader).await?
                 }
-                SqsTrigger::TRIGGER_TYPE => {
-                    let sqs_trigger =
-                        build_trigger::<SqsTrigger>(app.clone(), app_source.clone()).await?;
+                SQS_TRIGGER_TYPE => {
                     info!(" >>> running spin sqs trigger");
```

Comment:
suggest: These `info!()` could be moved to `run()` in `trigger.rs `

Reply:
Hmmm... the `run` function is generic and handles all of the triggers. We could move the message into that function but it would likely look like ">>> running trigger \`spin_trigger_http::HttpTrigger\`" which I'm not sure is better.

Reply:
Oh okay, I thought you can do `T::TYPE` to get the type of the trigger in `run()` function.